crates/rome_diagnostics/src/v2/display/diff.rs

```diff
@@ -0,0 +1,696 @@
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    io, slice,
+};
+
+use rome_console::{fmt, markup, MarkupElement};
+use rome_text_edit::{ChangeTag, CompressedOp, TextEdit};
+
+use super::frame::{
+    calculate_print_width, print_invisibles, text_width, IntoIter, OneIndexed,
+    PrintInvisiblesOptions, CODE_FRAME_CONTEXT_LINES,
+};
+
+const MAX_PATCH_LINES: usize = 150;
+
+pub(super) fn print_diff(fmt: &mut fmt::Formatter<'_>, diff: &TextEdit) -> io::Result<()> {
+    let mut modified_lines = BTreeSet::new();
+    let mut inserted_lines = BTreeMap::new();
+    let mut before_line_to_after = BTreeMap::new();
+
+    let mut before_line = OneIndexed::MIN;
+    let mut after_line = OneIndexed::MIN;
+
+    for (i, op) in diff.iter().enumerate() {
+        let op = match op {
+            CompressedOp::DiffOp(op) => op,
+            CompressedOp::EqualLines { line_count } => {
+                let is_first_op = i == 0;
+                for i in 0..=line_count.get() {
+                    // Don't increment the first line if we are the first tuple marking the beginning of the file
+                    if !(is_first_op && i == 0) {
```

Comment:
```suggestion
                    if !is_first_op {
```

because `is_first_op` already asserts that `i == 0`

Reply:
... ah no, this are different `i`. I think it would help readability to either rename the inner or outer `i` even if it's just naming them `i`, and `j`